pkg/loadbalancer/redirectpolicy/controller.go

```diff
@@ -322,13 +322,18 @@ func (c *lrpController) updateRedirects(wtxn writer.WriteTxn, ws *statedb.WatchS
 					}
 					continue
 				}
+				//if we only have one frontend mapping, we dont need the frontend port name so it will not check the port name in the backend ports
+				if len(lrp.FrontendMappings) == 1 {
+					feM.fePort = lb.FEPortName("")
+				}
 				_, err := c.p.Writer.UpsertFrontend(
 					wtxn,
 					lb.FrontendParams{
 						Address:     feM.feAddr,
 						Type:        lb.SVCTypeLocalRedirect,
 						ServiceName: lrpServiceName,
 						ServicePort: feM.feAddr.Port(),
+						PortName:    feM.fePort,
```

Comment:
Can we do this instead of overwriting the `feM.fePort `

```
PortName: func() string {
					if len(lrp.FrontendMappings) > 1 {
						return feM.fePort
					}
					return lb.FEPortName("")
				}(),
```

Reply:
I can definitely change this but I have questions/concerns.

Do we want to offer the users that the frontend not need to match the backend user experience when there is only one frontend?

If so, I think we can just keep the test this way https://github.com/cilium/cilium/blob/main/cilium-cli/connectivity/builder/manifests/local-redirect-policy.yaml#L11-L19 since that's kind of check from connectivity test perspective.

If we want users to keep the frontend/backend port the same even there is one frontend, we dont have to write the function at all, since it will just pass the frontend port name. Then in the writer, it will check the if that's correct.